src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+            Assert.True(IO.Directory.Exists(FullPathToNewDirectory))
+            Assert.True(IO.File.Exists(testFileSource))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub DeleteDirectory_Directory_UIOption_RecycleOption()
+
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub DeleteDirectory_Directory_UIOption_RecycleOption_UICancelOption()
+
+        End Sub
+
+        <Fact>
+        Public Sub DeleteFile_File()
+            Dim TestBase As New FileIOTestBase
+            Dim testFileSource As String = CreateTestFile(TestBase, SourceData)
+
+            Assert.True(IO.File.Exists(testFileSource))
+            FileSystem.DeleteFile(testFileSource)
+            Assert.False(IO.File.Exists(testFileSource))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub DeleteFile_File_UIOption_RecycleOption()
+
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub DeleteFile_File_UIOption_RecycleOption_UICancelOption()
+
+        End Sub
+
+        <Fact>
+        Public Sub DirectoryExists_directory()
+            Dim TestBase As New FileIOTestBase
+            Dim TestDirectory As String = TestBase.TestDirectory()
+            Assert.True(FileSystem.DirectoryExists(TestDirectory))
+            Assert.False(FileSystem.DirectoryExists(IO.Path.Combine(TestDirectory, "NewDirectory")))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub FileExists_File()
+            Dim TestBase As New FileIOTestBase
+            Dim testFileSource As String = CreateTestFile(TestBase, SourceData)
+            Assert.True(FileSystem.FileExists(testFileSource))
+            FileSystem.FileExists(testFileSource)
+            IO.File.Delete(testFileSource)
+            Assert.False(FileSystem.FileExists(testFileSource))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub FindInFiles_Directory_ContainsText_IgnoreCase_SearchOption_FileWildcards()
+
+        End Sub
+
+        <Fact>
+        Public Sub FindInFiles_Directory_ContainsText_IgnoreCase_SearchType_SearchOption()
+
+        End Sub
+
+        <Fact>
+        Public Sub GetDirectories_Directory()
+            Dim TestBase As New FileIOTestBase
+            Dim DirectoryList As ReadOnlyCollection(Of String) = FileSystem.GetDirectories(TestBase.TestDirectory)
+            Assert.True(DirectoryList.Count = 0)
+            For i As Integer = 0 To 5
+                IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory{i}"))
+            Next
+            DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory)
+            Assert.True(DirectoryList.Count = 6)
+            For i As Integer = 0 To 5
+                Assert.Equal(DirectoryList(i), IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory{i}"))
```

Comment:
how are the files ordered in this list ?

Reply:
Alphabetically and by time, This passes on Windows it may not on Linux which is why I did this PR to Test on Linux/Mac.